### PR TITLE
uast: unary operators

### PR DESCRIPTION
**Pull request by juanjux**
I think this cover the most important unary operators. There is also the option of composing them using the Binary[Aritmetic|Bitwise]Operators (changing their name to just operators) but some like takeaddress or dereference doesn't have analogue ones there.

### CODE REVIEW

File: uast/uast.go [hunk 1]
```diff
@@ -30,6 +30,28 @@ const (
 	Expression
 	Statement
 
+	// Unary operators
+	// ++x
+	UnaryPreIncrement
```
Comment (smola) [inline uast/uast.go]: Maybe it's better to use naming: such as `OpPreIncrement`?

File: uast/uast.go [hunk 2]
```diff
@@ -30,6 +30,28 @@ const (
...
+	// x++
+	UnaryPostIncrement
+	// --x
+	UnaryPreSubstract
```
Comment (smola) [inline uast/uast.go]: `Substract` could be replaced with `Decrement`, to not confuse it with binary substraction operation.

File: uast/uast.go [hunk 3]
```diff
@@ -30,6 +30,36 @@ const (
...
+	// Unary operators. These will have a single child node that will apply the operator over it.
+	// TODO: use BooleanNot to implement the UnaryNot?
+
+	// OpPreIncrement increments in place the value before it is evaluated. It's
+	// typical of C-inspired languages (e. g. ++x).
+	OpPreIncrement
+	// OpPostIncrement increments in place the value after it is evaluated. It's
+	// typical of C-inspired languages (e. g. x++).
+	OpPostIncrement
+	// OpPreIncrement decrement in place the value before it is evaluated. It's
```
Comment (smola) [inline uast/uast.go]: s/OpPreIncrement/OpPreDecrement

**Comment by juanjux** [comment]
Fixed.

**Pull request merged**